### PR TITLE
QR code login feature

### PR DESCRIPTION
**Pull request by JorgeMucientes**
This PR implementers the embedded QR code scanner code. It uses [Android ML learning kit](https://developers.google.com/ml-kit/vision/barcode-scanning/android) for it. The approach followed is using [Google Play Service](https://developers.google.com/ml-kit/vision/barcode-scanning/android#:~:text=If%20you%20choose%20to%20use%20the%20model%20in%20Google%20Play%20Services)s and downloading the ML kit on demand after app installation to avoid increasing .aab size

Please **DON'T PANIC**, I know the PR is huge, as this project required to add this embedded QR code in a different module. I picked up where @0nko left it from a hack week project. Thus, it was hard to split the PR into smaller ones. I've removed as much code as possible and only left what was needed. "Luckily" most of the file changes are small changes related to adding a new module to the project, or new resources and simple layouts. 

### Testing instructions
To test: 

- First request a magic link/QR code email
- Click on "Continue with WordPress.com" 
- Click on "Scan QR code to login" -> Check `login_with_qr_code_button_tapped, Properties: {"flow":"wordpress_com","is_debug":true}` is tracked. 
- Check that permission is requested (reject permission and retry again, grating permission) 
- Click on "Help" button inside QR code scanner screen and check it opens help screen. 
- Navigate back and scan the QR code. Verify the app logs in correctly. 
- Verify that when the code is scanned the following event is tracked: `login_with_qr_code_scanned, Properties: {"is_debug":true}`

See the screen recording for the expected behaviour: 

https://user-images.githubusercontent.com/2663464/194370715-7f5102a9-2a7f-4d81-9535-5bec572d1875.mp4

### CODE REVIEW

File: libs/barcode/build.gradle [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+dependencies {
+    implementation "androidx.core:core-ktx:$coreKtxVersion"
+    implementation "androidx.annotation:annotation:1.4.0"
+    implementation "androidx.interpolator:interpolator:1.0.0"
+    implementation "com.google.android.material:material:$materialVersion"
+    implementation 'androidx.preference:preference-ktx:1.2.0'
+
+    // Barcode model
+    implementation 'com.google.android.gms:play-services-mlkit-barcode-scanning:18.0.0'
```
Comment (0nko) [inline libs/barcode/build.gradle]: Seems like there is a newer version available - 18.1.0

**Review by 0nko — COMMENTED** [review]
Works really great, nice work, @JorgeMucientes! 🎖️ I had no problem with the scanning, permissions and logging in.

Just one observation - after I entered my email address and requested a magic link, I think it would make sense to add a button for QR scanner to this screen (instead of having to go back):

image

WDYT?

**Comment by JorgeMucientes** [comment]
Thanks for the review @0nko 🙏🏼 

I've updated the ML Kit version to the latests version and updated `RELEASE-NOTES.TXT` so this is ready for merging now.
 
Regarding the suggested idea, I think it's great. As discussed over Slack let's add that change on a next iteration on `10.8` so we can merge this changes in `v10.7`

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**